mitodl/teachersportal
# Removed django-oscar

#### What's this PR do?

Remove django-oscar and replaces its Product model with Course and Module.

This PR does not change the product API so unfortunately we aren't getting rid of all product terminology (see #253 to complete this). This PR should clean up all other references to products or UPCs or SKUs however.

This PR also adds admin forms to edit the Course and Module models. There is now a `live` boolean on the Course which must be activated in addition to the Module having a correct price.
#### Where should the reviewer start?

`portal/models.py`
#### How should this be manually tested?

Note: You will need to wipe the database to test this PR, it doesn't preserve old migrations or old data in general.

To test:
- Wipe your database. Create a superuser
- Set up OAuth credentials for CCXCon as before
- Either use a webhook to tell CCXCon to create a Course and Module, or do this manually in the Django Admin forms.
- Go into Django Admin forms and add a price to the Module, and make sure Course's `live` checkbox is checked.
- You should be able to view the course detail page, and go through the checkout flow as before.
#### What are the relevant tickets?

Fixed #187
